### PR TITLE
QNTM-1636: Fix load time error and add unit tests

### PR DESCRIPTION
**Pull request by ColinDayOrg**
### Purpose

This PR addresses an issue with loading a file that uses nodes from an imported library after restarting Dynamo. It also adds unit tests to cover QNTM-1636 functionality.

### Declarations

Check these if you believe they are true

- [X] The code base is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [X] User facing strings, if any, are extracted into `*.resx` files
- [X] All tests pass using the self-service CI.
- [X] Snapshot of UI changes, if any.
- [X] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@mjkkirschner 

### FYIs

@QilongTang 
@ramramps

### CODE REVIEW

File: src/DynamoCore/Models/DynamoModel.cs [hunk 1]
```diff
@@ -768,15 +768,24 @@ protected DynamoModel(IStartConfiguration config)
         /// does not already exist in the list.
         /// </summary>
         /// <param name="path"> The path to add.</param>
-        public bool AddPackagePath(string path)
+        /// <param name="file"> The file to add when importing a library.</param>
+        public bool AddPackagePath(string path, string file)
```
Comment (ramramps) [inline src/DynamoCore/Models/DynamoModel.cs]: can it be `public bool AddPackagePath(string path, string file="")` ? i.e adding default param, so the other places doesn't need a change.

File: test/DynamoCoreTests/DSLibraryTest.cs [hunk 2]
```diff
@@ -73,6 +73,26 @@ public void TestLoadDllFileFailure()
 
         [Test]
         [Category("UnitTests")]
+        public void TestLoadDllFileSuccess()
+        {
+            bool libraryLoaded = false;
+
+            libraryServices.LibraryLoaded += (sender, e) => libraryLoaded = true;
```
Comment (mjkkirschner) [inline test/DynamoCoreTests/DSLibraryTest.cs]: I think we should unsubscribe this when the test ends.

File: test/DynamoCoreTests/DSLibraryTest.cs [hunk 3]
```diff
@@ -94,6 +114,53 @@ public void TestLibraryAcrossSessions()
         }
 
         [Test]
+        [Category("UnitTests")]
+        public void TestDllLibraryAcrossSessions()
+        {
+            bool libraryLoaded = false;
+            libraryServices.LibraryLoaded += (sender, e) => libraryLoaded = true;
```
Comment (mjkkirschner) [inline test/DynamoCoreTests/DSLibraryTest.cs]: same here - unsubscribe.

**Comment by mjkkirschner** [comment]
@ColinDayOrg looks good to me besides the three comments.

**Comment by mjkkirschner** [comment]
LGTM

File: src/DynamoCore/Models/DynamoModel.cs [hunk 4]
```diff
@@ -768,15 +768,24 @@ protected DynamoModel(IStartConfiguration config)
         /// does not already exist in the list.
         /// </summary>
         /// <param name="path"> The path to add.</param>
-        public bool AddPackagePath(string path)
+        /// <param name="file"> The file to add when importing a library.</param>
+        public bool AddPackagePath(string path, string file = "")
```
Comment (QilongTang) [inline src/DynamoCore/Models/DynamoModel.cs]: Can we use string.Empty instead of ""? They are same but just for better readability

**Comment by QilongTang** [comment]
LGTM except for one comment

File: src/DynamoCore/Models/DynamoModel.cs (hunk 4 above)
Comment (ColinDayOrg) [reply src/DynamoCore/Models/DynamoModel.cs]: I gave it a try but apparently default arguments are required to be compile-time constants, and string.Empty is not a compile-time constant, so I don't think it is possible in this case.

Comment (QilongTang) [reply src/DynamoCore/Models/DynamoModel.cs]: Thanks for the trial, I wasn't aware of this good info before. Then LGTM

**Pull request merged**